redis/src/types.rs

```diff
@@ -778,13 +778,21 @@ impl fmt::Debug for RedisError {
     }
 }
 
-pub(crate) enum RetryMethod {
+/// What method should be used if retrying this request.
+pub enum RetryMethod {
+    /// Create a fresh connection.
     Reconnect,
+    /// Don't retry, this is a permanent error.
     NoRetry,
+    /// Retry immediately, this doesn't require a wait since it's an IO level issue.
     RetryImmediately,
+    /// Retry after sleeping to avoid overloading the external service.
     WaitAndRetry,
+    /// The key has moved to a different node but we have to ask which node.
     AskRedirect,
```

Comment:
I'm also okay if you want to hide these behind feature flags?  I just didn't because the actual ErrorKinds themselves aren't (unlike Json which is).